src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseEventEnablingService.java

```diff
@@ -24,4 +33,24 @@ public Boolean isEventEnabled(String enablingCondition,
         }
         return Boolean.TRUE;
     }
+
+    public Boolean isEventEnabled(String enablingCondition,
+                                  List<CaseViewField> caseViewFields) {
+        if (StringUtils.isNotEmpty(enablingCondition)) {
+            Map<String, Object> caseViewFieldData = caseViewFields.stream()
+                .filter(injectedDataCaseViewField -> injectedDataCaseViewField.getId().startsWith("[INJECTED_DATA"))
+                .collect(
+                    Collectors.toMap(
+                        caseViewField -> replaceDotDelimiter(caseViewField.getId()),
+                        caseViewField -> TextNode.valueOf((String) caseViewField.getValue())));
+
+            return !caseViewFieldData.isEmpty()
+                && this.enablingConditionParser.evaluate(replaceDotDelimiter(enablingCondition), caseViewFieldData);
+        }
+        return Boolean.TRUE;
+    }
+
+    private String replaceDotDelimiter(String delimitedString) {
+        return delimitedString.replace(DOT_DELIMITER, DOT_DELIMITER_REPLACEMENT);
```

Comment:
Another observation to do with chaining e.g. `ComplexField.EmailField = "email.value@has.a.dot.com" AND [INJECTED_DATA.OtherField] = "B"` - we should probably replace `INJECTED_DATA.` with `INJECTED_DATA__` (or similar), otherwise any complex fields being referenced, or dots in the values, will also be replaced.